contracts/test/PermanentStorage.t.sol

```diff
@@ -0,0 +1,428 @@
+            vm.expectEmit(false, false, false, true);
+            emit SetRelayerValid(_relayer, valid);
+        }
+        permanentStorage.setRelayersValid(DEFAULT_RELAYERS, DEFAULT_RELAYER_VALIDS);
+
+        for (uint256 i = 0; i < DEFAULT_RELAYERS.length; i++) {
+            address _relayer = DEFAULT_RELAYERS[i];
+            assertTrue(permanentStorage.isRelayerValid(_relayer));
+        }
+    }
+
+    /********************************************
+     *          Test: setCurvePoolInfo          *
+     ********************************************/
+
+    function testCannotSetCurvePoolInfoWithoutPermission() public {
+        vm.expectRevert("PermanentStorage: has no permission");
+        vm.prank(user);
+        permanentStorage.setCurvePoolInfo(
+            DEFAULT_CURVE_POOL_ADDRESS,
+            DEFAULT_CURVE_POOL_UNDERLYING_COINS,
+            DEFAULT_CURVE_POOL_COINS,
+            DEFAULT_CURVE_SUPPORT_GET_DX
+        );
+    }
+
+    function testCannotGetCurvePoolInfoIfNotSet() public {
+        address underlyingCoinA = DEFAULT_CURVE_POOL_UNDERLYING_COINS[0];
+        address underlyingCoinB = DEFAULT_CURVE_POOL_UNDERLYING_COINS[1];
+        vm.expectRevert("PermanentStorage: invalid pair");
+        permanentStorage.getCurvePoolInfo(DEFAULT_CURVE_POOL_ADDRESS, underlyingCoinA, underlyingCoinB);
+
+        address coinA = DEFAULT_CURVE_POOL_COINS[0];
+        address coinB = DEFAULT_CURVE_POOL_COINS[1];
+        vm.expectRevert("PermanentStorage: invalid pair");
+        permanentStorage.getCurvePoolInfo(DEFAULT_CURVE_POOL_ADDRESS, coinA, coinB);
+    }
+
+    function testSetCurvePoolInfo() public {
+        bytes32 storageId = permanentStorage.curveTokenIndexStorageId();
+        permanentStorage.setPermission(storageId, address(this), true);
+
+        vm.expectEmit(false, false, false, true);
+        emit SetCurvePoolInfo(DEFAULT_CURVE_POOL_ADDRESS, DEFAULT_CURVE_POOL_UNDERLYING_COINS, DEFAULT_CURVE_POOL_COINS, DEFAULT_CURVE_SUPPORT_GET_DX);
+        permanentStorage.setCurvePoolInfo(
+            DEFAULT_CURVE_POOL_ADDRESS,
+            DEFAULT_CURVE_POOL_UNDERLYING_COINS,
+            DEFAULT_CURVE_POOL_COINS,
+            DEFAULT_CURVE_SUPPORT_GET_DX
+        );
+
+        // Check underlying coin info
+        for (uint256 i = 0; i < DEFAULT_CURVE_POOL_UNDERLYING_COINS.length; i++) {
+            address underlyingCoinA = DEFAULT_CURVE_POOL_UNDERLYING_COINS[i];
+            for (uint256 j = i; j < DEFAULT_CURVE_POOL_UNDERLYING_COINS.length; j++) {
+                address underlyingCoinB = DEFAULT_CURVE_POOL_UNDERLYING_COINS[j];
+                (int128 takerAssetIndex, int128 makerAssetIndex, uint16 swapMethod, bool supportGetDx) = permanentStorage.getCurvePoolInfo(
+                    DEFAULT_CURVE_POOL_ADDRESS,
+                    underlyingCoinA,
+                    underlyingCoinB
+                );
+                assertGt(takerAssetIndex, 0);
+                assertGt(makerAssetIndex, 0);
```

Comment:
Q: Not sure why the indexes are all 0, I thought they would be increasing?

Reply:
It would more complex to check the exact index so I use `assert greater than` instead to at least check that they are written to and hence not returning a default value `0`.

Reply:
Oh I miss the `Gt` keyword 🤦‍♂️ No problem 💪